src/manifest.c

```diff
@@ -616,16 +616,27 @@ struct list *create_update_list(struct manifest *current, struct manifest *serve
 		file = list->data;
 		list = list->next;
 
-		/* Skip files that aren't updated */
-		fullname = mk_full_filename(path_prefix, file->filename);
-		if (fullname == NULL) {
-			abort();
-		}
-		if (verify_file(file, fullname)) {
+		if (file_found_in_manifest(current, file)) {
```

Comment:
Good solution but needs to use peers because this is an O(N^2) operation otherwise (file_has_different_hash_in_manifest loops over the full manifest file list).